lib/tapioca/sorbet_ext/generic_name_patch.rb

```diff
@@ -20,31 +21,47 @@ def [](*types)
         Tapioca::Runtime::GenericTypeRegistry.register_type(constant, types)
       end
 
-      def type_member(variance = :invariant, fixed: nil, lower: T.untyped, upper: BasicObject)
+      def type_member(variance = :invariant, fixed: nil, lower: nil, upper: nil, &blk)
         # `T::Generic#type_member` just instantiates a `T::Type::TypeMember` instance and returns it.
         # We use that when registering the type member and then later return it from this method.
+        hash = if blk
```

Comment:
Instead of checking if the block is present, should we make the sorbet supports check here?

Reply:
This is the processing side of the equation, which can be more relaxed with respect to how bound parameters are passed. Ideally we would like to support mixed usage of bounds being passed as arguments or inside a block, since the `pipeline_spec` is testing exactly that.

For generating side of the equation, we need to be more strict so that the Sorbet version we run against can properly process the generated code.

This is all following the old internet adage of "be liberal with what you accept and conservative with what you send".